### ISSUE TITLE
Next/link triggers [HMR] bundle rebuilding but does not change page

### ISSUE THREAD

**Issue by ajhool** [zeit/next.js#6552]
# Bug report

## To Reproduce

I'm working on reproducing, but next.js provides zero information about what's happening, so it's nearly impossible for me to actually identify where the problem is. All I know is that my code isn't working.

## Expected behavior

Clicking this component should redirect to the about page.
```
<Link href="/about">About</Link>
```

A meta issue: If it does not redirect to the about page, next.js should provide some debug output to explain why it's not working. It's hard to isolate a bug when the entire HMR system is blackboxed.

## System information
 - macOS
 - chrome
 - Next.js v 8.0.0

## Additional context

I'm using a custom server and clean routes. The clean route pages do load properly, but the other pages in the "/pages" directory do not load properly. If I navigate directly to the url (eg. /about), the About page does load.

This might be related to #6317 and #6497

**Comment by Yzrsah** [comment]
1) Create a series of Next testing targets:

- For each test target, extend functionality of the previous target

`NextWithServer` ->
`NextWithServerPWA` ->
`NextEnvCustomPWA` ->
`NextEnvCustomPWAFirestore` ->
...
`NextConfigStyledPWACleanStatic` ->
...
`NextSuperSite` ->
`NextMegaSite`

2) Using [next-compose-plugins](https://www.npmjs.com/package/next-compose-plugins)

- Import from a hierarchy of `compose.config.js` and `compose.babelrc.js` files in order to add functionality from each simpler setup

3) In your test directory, make tests which copy:

`Views`
`Components`
`BasicSiteBootstrap`

For a set of the basic things your site relies on to render a test page which has enough relevancy or similarity with the production pages.

- Create a script which allows you to test your test pages using each of the Next targets.

### . :

Basically, it is actually our own fault, and we can track down all issues by creating reproductions for issues with this amazing free software.

When we ask Zeit to worry about issues which we *could* have made a reproduction for, it is more difficult for them to have free time to work on bringing the great enhancements that Next needs.

**Comment by Yzrsah** [comment]
1) Create a series of Next testing targets:

- For each test target, extend functionality of the previous target

`NextWithServer` ->
`NextWithServerPWA` ->
`NextEnvCustomPWA` ->
`NextEnvCustomPWAFirestore` ->
...
`NextConfigStyledPWACleanStatic` ->
...
`NextSuperSite` ->
`NextMegaSite`

2) Using [next-compose-plugins](https://www.npmjs.com/package/next-compose-plugins)

- Import from a hierarchy of `compose.config.js` and `compose.babelrc.js` files in order to add functionality from each simpler setup

3) In your test directory, make tests which copy:

`Views`
`Components`
`BasicSiteBootstrap`

For a set of the basic things your site relies on to render a test page which has enough relevancy or similarity with the production pages.

- Create a script which allows you to test your test pages using each of the Next targets.

### . :

Basically, it is our own fault, and we can track down all issues by creating reproductions for issues with this amazing free software.

When we ask Zeit to worry about issues which we *could* have made a reproduction for, it is more difficult for them to have free time to work on bringing the great enhancements that Next needs.

**Comment by ajhool** [comment]
GitHub issues are also a way to find other people who are having similar problems. Reproductions are hard for a complex system and isolating this particular issue is nontrivial. I'm curious if other people are seeing it as an issue so we can work through the problem together. I do pay Zeit for their software.

**Comment by Yzrsah** [comment]
@ajhool I've been going through what you're going through. It is a non-stop with this kind of thing when it comes to living at HEAD. But Next actually reduces these kinds of issues significantly. I realized that I have to solve the issues on my own by creating verified+tested targets, because that means I have control over the situation.

If you have added 25 packages but you're not sure which one introduced breakage, then that is a loss of control over the code.

"Should I post an issue about this and try to convince people to understand? Nah screw it, let's just get down to business and test it using the verified working test targets we have made."

**Comment by Yzrsah** [comment]
@ajhool I've been going through what you're going through. It is a non-stop with this kind of thing when it comes to living at HEAD. But Next actually reduces these kinds of issues significantly. I realized that I have to solve the issues on my own by creating verified+tested targets, because that means I have control over the situation.

If you have added 25 packages and a bunch of custom Webpack rules but you're not sure which one introduced breakage, then that is a loss of control over the code.

"Should I post an issue about this and try to convince people to understand? Nah screw it, let's just get down to business and test it using the verified working test targets we have made."

**Comment by Yzrsah** [comment]
@ajhool I've been going through what you're going through. It is a non-stop with this kind of thing when it comes to living at HEAD. But Next actually reduces these kinds of issues significantly. I realized that I have to solve the issues on my own by creating verified+tested targets, because that means I have control over the situation.

If you have added 25 packages and a bunch of custom Webpack rules but you're not sure which one introduced breakage, then that is a loss of control over the code.

I often think "Should I post an issue about this and try to convince people to understand? Nah screw it, let's just get down to business and tear things up test it using the verified working test targets we have made."

You have a good idea about posting on GitHub to connect with people who have similar issues. But if you can make reproductions from simpler tested setups, at least there is always something that you can do instead of waiting.

**Comment by ajhool** [comment]
> You have a good idea about posting on GitHub to connect with people who have similar issues. But if you can make reproductions from simpler tested setups, at least there is always something that you can do instead of waiting

I am posting this issue while simultaneously working on a solution

**Comment by Yzrsah** [comment]
Did you tear up your setup?

**Comment by Yzrsah** [comment]
> I am also only posting this issue after having spent a substantial amount of time try to identify the cause

Did you tear up your setup?

**Comment by ajhool** [comment]
Yes. I have several complex integrations, which include Apollo, Amplify, and Material-UI. I'm honing in on the problem being an Amplify SSR issue and I'm honing in on a workaround

**Comment by timneutkens** [comment]
Can you post next.config.js, and package.json

**Comment by adueck** [comment]
@ajhool I am having the exact same problem. It only appeared for me *when I upgraded to Next 8*. That was the only change I made.  

This is baffling for me because all my `<Link>`s work fine, except for my links to one particular type of page. (I am using clean routing with a custom server, but it's also broken without clean routing)

**Comment by adueck** [comment]
@ajhool Ok, I figured out what was breaking it for me.  

There was one page where I was experiencing the same problem with the `<Link>` not switching to the page. On that page I had a **css file import**. Removing that fixed it for me.  

Here is my `next.config.js`

```
const withPlugins = require("next-compose-plugins");
const withCSS = require('@zeit/next-css');
const withTypescript = require('@zeit/next-typescript')
const webpack = require('webpack');

module.exports = withPlugins([withCSS, withTypescript], {
      webpack: (config) => {
          config.plugins.push(
              new webpack.ProvidePlugin({
                  '$': 'jquery',
                  'jQuery': 'jquery',
              })
          );
          config.module.rules.push({
            test: /\.(eot|woff|woff2|ttf|svg|png|jpg|gif)$/,
            use: {
              loader: 'url-loader',
              options: {
                limit: 100000,
                name: '[name].[ext]'
              }
            }
          });
          return config
      }
});
```

**Comment by timneutkens** [comment]
Duplicate of https://github.com/zeit/next-plugins/issues/282

**Comment by DominicTobias-b1** [comment]
Mine changes page but often rebuilds the bundle when it does 🤔
